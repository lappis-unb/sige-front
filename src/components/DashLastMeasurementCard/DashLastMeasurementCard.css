.readings {
  width: 100%;
  text-align: center;
}

.h4 {
  font-weight: 600;
  color: #339cff;
  line-height: 36px;
  size: 24px;
  text-transform: uppercase;
}

.reading-measurement {
  font-size: 0.8em;
}

td {
  font-size: 20px;
}

@media screen and (max-width: 800px) {
  .last-card-info {
    margin-bottom: 20px;
  }
}
