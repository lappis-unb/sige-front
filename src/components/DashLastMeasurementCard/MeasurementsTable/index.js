import MeasurementsTable from './MeasurementsTable.vue';
export default MeasurementsTable;