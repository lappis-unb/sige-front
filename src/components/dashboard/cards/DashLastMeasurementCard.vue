<template>
  <q-card
    flat
    class="card-bg">

    <q-inner-loading dark :showing="transductor.name === undefined">
      <q-spinner-ios size="50px" color="grey-4" thickness="7"/>
    </q-inner-loading>

    <q-card-section v-if="transductor.name" class="q-pt-xs q-pb-none text-center card-title">
      {{ transductor.name[0].toUpperCase() + transductor.name.slice(1) }}
    </q-card-section>

    <q-card-section v-if="rtm.length !== 0" class="q-pt-none q-pb-xs">
      <table class="readings">
        <tr class="row">
          <th class="col h4">
            Tensão
          </th>
          <th class="col h4">
            Corrente
          </th>
          <th class="col h4">
            Potência
          </th>
          <th class="col h4">
            Últimas 72h
          </th>
        </tr>

        <tr class="row">
          <td class="col">A - {{ rtm.tension_a }}V</td>
          <td class="col">A - {{ rtm.current_a }}A</td>
          <td class="col">A - {{ rtm.active_power_a }}W</td>
          <td class="col">
            {{ 0 }} <q-icon :style="{opacity: 0.5}" :name="'img:statics/icons/ic_ocorrencia_critica_mono.svg'"/>
          </td>
        </tr>

        <tr class="row">
          <td class="col">B - {{ rtm.tension_b }}V</td>
          <td class="col">B - {{ rtm.current_b }}A</td>
          <td class="col">B - {{ rtm.active_power_b }}kVAr</td>
          <td class="col">
            {{ 0 }} <q-icon :style="{opacity: 0.5}" :name="'img:statics/icons/ic_ocorrencia_precaria_mono.svg'"/>
          </td>
        </tr>

        <tr class="row">
          <td class="col">C - {{ rtm.tension_c }}V</td>
          <td class="col">C - {{ rtm.current_c }}A</td>
          <td class="col">T - {{ rtm.active_power_c }}kVa</td>
          <td class="col"></td>
        </tr>
      </table>
    </q-card-section>

    <q-card-section v-if="transductor.name" class="q-pt-xs">
      <h6 class="text-center" style="color: rgba(255, 255, 255, 0.6)">
        Última medida não disponível
      </h6>
    </q-card-section>
  </q-card>
</template>

<script>
import HTTP from '../../../services/masterApi/http-common'

export default {
  name: 'DashLastMeasurementCard',

  data () {
    return {
      rtm: [],
      transductor_occurences: {}
    }
  },

  props: {
    transductor: {
      // type: Object,
      required: true
    }
  },

  methods: {
    getLastMeasurement () {
      HTTP
        .get(`/realtime-measurements/?serial_number=${this.transductor.serial_number}`)
        .then((res) => {
          this.rtm = res.data
        })
        .catch((err) => {
          console.error(err)
        })
    },
    getTransductorsLast72h () {
      HTTP
        .get(`/occurences/?type=period&serial_number=${this.transductor.serial_number}`)
        .then((res) => {
          this.transductor_occurences = res.data
        })
        .catch((err) => {
          console.error(err)
        })
    },

    async getApiInfo () {
      await this.getLastMeasurement()
      await this.getTransductorsLast72h()
    }
  },

  mounted () {
    this.getApiInfo()
  },

  async beforeUpdate () {
    this.getApiInfo()
  }
}
</script>

<style lang="scss" scoped>
  .readings{
    width: 100%;
    text-align: center;
  }

  .h4 {
    font-weight: 600;
    color: #339cff;
    line-height: 36px;
    size: 24px;
    text-transform: uppercase;
  }

  td {
    font-size: 20px;
  }

</style>
