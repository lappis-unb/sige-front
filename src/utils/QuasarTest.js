import Vuex from 'vuex'
import Vue from 'vue'
import { createLocalVue, shallowMount, mount } from '@vue/test-utils';
import Quasar from 'quasar';

class QuasarTest {
  
  _localVue;

  constructor(){
    this.setUpQuasar()
  }
  
  setUpQuasar() {
    this._localVue = createLocalVue()
    Vue.use(Quasar)
    this._localVue.use(Vuex)
    this._localVue.use(Quasar)
  }

  shallowMount (Component, propsData = {}) {
    const localVue = this._localVue;
    return shallowMount(Component, {
      localVue,
      propsData
    });
  }

  mount (Component, propsData = {}) {
    const localVue = this._localVue;
    return mount(Component, {
      localVue,
      propsData
    });
  }
}

export default new QuasarTest();