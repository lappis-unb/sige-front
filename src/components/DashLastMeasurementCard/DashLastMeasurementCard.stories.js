import DashLastMeasurementCard from '.'

export default {
  title: 'Dash Last Measurement Card',
  component: DashLastMeasurementCard,
  argTypes: {
    transductor: {type: Object},
    transductor_occurences: {type: Object},
    realTimeMeasurements: {type: Array},
    measurementsCallback: {type: Function}
  }
}

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { DashLastMeasurementCard },
  template: '<dash-last-measurement-card v-bind="$props"/>'
})

export const Default = Template.bind({})
Default.args = {
  transductor: {
    name: 'FGA - Gama'
  },
  transductor_occurences: {},
  realTimeMeasurements: [{voltage_a: 134 ,voltage_b: 32, voltage_c: 131}],
  measurementsCallback: () => {},
  reports: {
    precarious_tension: [1, 23, 4, 6],
    transductor_connection_fail: [1, 23, 4, 6],
    slave_connection_fail: [1, 23, 4, 6],
    critical_tension: [1],
    phase_drop: [1]
  }
}
