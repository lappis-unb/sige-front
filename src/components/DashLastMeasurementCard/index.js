import DashLastMeasurementCard from './DashLastMeasurementCard.vue';
export default DashLastMeasurementCard;