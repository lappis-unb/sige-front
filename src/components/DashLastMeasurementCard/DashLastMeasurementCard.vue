<template>
  <q-card
    flat
    class="card-bg last-card-info">

    <q-inner-loading dark :showing="transductor.name === undefined">
      <q-spinner-ios size="50px" color="grey-4" thickness="7"/>
    </q-inner-loading>

    <q-card-section v-if="transductor.name" class="q-pt-xs q-pb-none text-center card-title">
      {{ transductor.name[0].toUpperCase() + transductor.name.slice(1) }}
    </q-card-section>

    <q-card-section v-if="realTimeMeasurements.length !== 0" class="q-pt-none q-pb-xs">
      <table class="readings row">
        <measurements-table 
          :title="'Tensão'" 
          :tableObject="{A:'voltage_a',B:'voltage_b',C:'voltage_c'}"
          :realTimeMesure="realTimeMeasurements"
          :unit="'V'"
        />
        <measurements-table 
          :title="'Corrente'" 
          :tableObject="{A:'current_a',B:'current_b',C:'current_c'}"
          :realTimeMesure="realTimeMeasurements"
          :unit="'A'"
        />
        <measurements-table 
          :title="'Potência'" 
          :tableObject="{Ativa:'total_active_power',Reativa:'total_reactive_power',Total:'total_power_factor'}"
          :realTimeMesure="realTimeMeasurements"
          :unit="['W', 'kVAr', 'kVa']"
        />
        <measurements-table 
          :title="'Últimas 72h'" 
          :tableObject="{A: this.countCriticalEvents(), B: this.countWarningEvents()}"
          :icons="[`${IMAGE_PATH}critica_mono.svg`, `${IMAGE_PATH}precaria_mono.svg`]"
        />
      </table>
    </q-card-section>

    <q-card-section v-if="transductor.name && realTimeMeasurements.length === 0" class="q-pt-xs">
      <h6 class="text-center" style="color: rgba(255, 255, 255, 0.6)">
        Última medida não disponível
      </h6>
    </q-card-section>
  </q-card>
</template>

<script>
import MeasurementsTable from './MeasurementsTable'

export default {
  components: { MeasurementsTable },
  name: 'DashLastMeasurementCard',

  props: {
    transductor: {
      type: Object
    },
    transductor_occurences: {
      type: Object
    },
    realTimeMeasurements: {
      type: Array
    },
    measurementsCallback: {
      type: Function
    }
  },

  data() {
    return {
      IMAGE_PATH: 'img:statics/ic_ocorrencia_',
    }
  },

  watch: {
    transductor: function () {
      this.measurementsCallback()
    }
  },

  mounted () {
    this.measurementsCallback()
  },

  methods: {
    countWarningEvents () {
      if (this.transductor_occurences.count) {
        const { slave_connection_fail, transductor_connection_fail } = this.transductor_occurences;
        return slave_connection_fail.length + transductor_connection_fail.length;
      }
      return 0
    },

    countCriticalEvents () {
      if (this.transductor_occurences.count) {
        const {critical_tension, phase_drop, precarious_tension} = this.transductor_occurences;
        return critical_tension.length + phase_drop.length + precarious_tension.length;
      }
      return 0
    },
  }
}
</script>

<style>
  @import "./DashLastMeasurementCard.css";
</style>
