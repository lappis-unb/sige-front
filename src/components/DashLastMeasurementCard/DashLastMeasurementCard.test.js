import DashLastMeasurementCard from './'
import QuasarTest from 'src/utils/QuasarTest';

const MOCKS = {
  transductor: {
    name: 'FGA - Gama'
  },
  transductor_occurences: {
    count: 1245,
    precarious_tension: [1, 23, 4, 6],
    transductor_connection_fail: [1, 23, 4, 6],
    slave_connection_fail: [1, 6, 4],
    critical_tension: [1],
    phase_drop: [1]
  },
  realTimeMeasurements: {
    voltage_a: 123.34 ,
    voltage_b: 32.312, 
    voltage_c: 131.3124,
    current_a: 113.34 ,
    current_b: 332.312, 
    current_c: 151.1124,
    total_active_power: 312.1,
    total_reactive_power: 3214.134,
    total_power_factor: 6541.23
  },
  measurementsCallback: jest.fn(),
}

function withWrapperArray(wrapperArray) {
  return {
    childSelectorHasText: (selector, str) => wrapperArray.filter(i => i.find(selector).text().match(str)),
    hasText: (str) => wrapperArray.filter(i => i.text().match(str)),
  }
}

describe('Dash Last Measurement Card', () => {
  describe('Teste de snapshot', () => {
    it("Componente formato Default", () => {
      const wrapper = QuasarTest.mount(DashLastMeasurementCard, MOCKS)
      expect(wrapper.element).toMatchSnapshot() 
    })
  })

  describe("Callback", ()=>{
    it('Função deve ser chamada assim que o componente for renderizado', () => {
      QuasarTest.mount(DashLastMeasurementCard, MOCKS);
      expect(MOCKS.measurementsCallback).toHaveBeenCalled();
    })
  })

  describe("Soma dos eventos", ()=>{
    it('O valor de eventos precarios deve ser a soma dos eventos do array', () => {
      const wrapper = QuasarTest.mount(DashLastMeasurementCard, MOCKS);
      const critical = wrapper.html().includes('6');
      const precary = wrapper.html().includes('7');
      expect(critical).toBeTruthy();
      expect(precary).toBeTruthy();
    })
  })
})
