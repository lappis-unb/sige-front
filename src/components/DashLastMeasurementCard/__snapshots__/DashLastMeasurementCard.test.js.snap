// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dash Last Measurement Card Teste de snapshot Componente formato Default 1`] = `
<div
  class="card-bg last-card-info q-card q-card--flat no-shadow"
>
  <transition-stub
    appear="true"
    name="q-transition--fade"
  />
   
  <div
    class="q-pt-xs q-pb-none text-center card-title q-card__section q-card__section--vert"
  >
    
    FGA - Gama
  
  </div>
   
  <div
    class="q-pt-none q-pb-xs q-card__section q-card__section--vert"
  >
    <table
      class="readings row"
    >
      <tr
        class="col-6 col-sm-6 col-md-6 col-lg-3 col-xl-3"
      >
        <span
          class="row"
        >
          <th
            class="col-12 h4"
          >
            
      Tensão
    
          </th>
           
          <td
            class="col-12 reading-measurement"
          >
            
      A: 123 V
      
            <!---->
          </td>
          <td
            class="col-12 reading-measurement"
          >
            
      B: 32 V
      
            <!---->
          </td>
          <td
            class="col-12 reading-measurement"
          >
            
      C: 131 V
      
            <!---->
          </td>
        </span>
      </tr>
       
      <tr
        class="col-6 col-sm-6 col-md-6 col-lg-3 col-xl-3"
      >
        <span
          class="row"
        >
          <th
            class="col-12 h4"
          >
            
      Corrente
    
          </th>
           
          <td
            class="col-12 reading-measurement"
          >
            
      A: 113 A
      
            <!---->
          </td>
          <td
            class="col-12 reading-measurement"
          >
            
      B: 332 A
      
            <!---->
          </td>
          <td
            class="col-12 reading-measurement"
          >
            
      C: 151 A
      
            <!---->
          </td>
        </span>
      </tr>
       
      <tr
        class="col-6 col-sm-6 col-md-6 col-lg-3 col-xl-3"
      >
        <span
          class="row"
        >
          <th
            class="col-12 h4"
          >
            
      Potência
    
          </th>
           
          <td
            class="col-12 reading-measurement"
          >
            
      Ativa: 312 W
      
            <!---->
          </td>
          <td
            class="col-12 reading-measurement"
          >
            
      Reativa: 3214 kVAr
      
            <!---->
          </td>
          <td
            class="col-12 reading-measurement"
          >
            
      Total: 6541 kVa
      
            <!---->
          </td>
        </span>
      </tr>
       
      <tr
        class="col-6 col-sm-6 col-md-6 col-lg-3 col-xl-3"
      >
        <span
          class="row"
        >
          <th
            class="col-12 h4"
          >
            
      Últimas 72h
    
          </th>
           
          <td
            class="col-12 reading-measurement"
          >
            
       6 
      
            <img
              aria-hidden="true"
              class="q-icon notranslate"
              role="presentation"
              src="statics/ic_ocorrencia_critica_mono.svg"
              style="opacity: 0.5;"
            />
          </td>
          <td
            class="col-12 reading-measurement"
          >
            
       7 
      
            <img
              aria-hidden="true"
              class="q-icon notranslate"
              role="presentation"
              src="statics/ic_ocorrencia_precaria_mono.svg"
              style="opacity: 0.5;"
            />
          </td>
        </span>
      </tr>
    </table>
  </div>
   
  <!---->
</div>
`;
